src/main/java/com/microsoftopentechnologies/windowsazurestorage/AzureStorageBuilder.java

```diff
@@ -53,48 +53,94 @@
 import jenkins.tasks.SimpleBuildStep;
 import net.sf.json.JSONObject;
 import org.acegisecurity.AccessDeniedException;
+import org.jenkinsci.Symbol;
 import org.kohsuke.stapler.AncestorInPath;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 import org.kohsuke.stapler.QueryParameter;
 import org.kohsuke.stapler.StaplerRequest;
 
 public class AzureStorageBuilder extends Builder implements SimpleBuildStep {
 
+    private final String storageCredentialId;
     private final transient String storageAccName;
     private final String downloadType;
-    private final String containerName;
-    private final String includeFilesPattern;
-    private final String excludeFilesPattern;
-    private final String downloadDirLoc;
-    private final boolean flattenDirectories;
-    private final boolean includeArchiveZips;
-    private final BuildSelector buildSelector;
-    private final String projectName;
-    private String storageCredentialId;
+    private String containerName = "";
+    private String includeFilesPattern = "";
+    private String excludeFilesPattern = "";
+    private String downloadDirLoc = "";
+    private boolean flattenDirectories;
+    private boolean includeArchiveZips;
+    private BuildSelector buildSelector;
+    private String projectName = "";
 
     private transient AzureCredentials.StorageAccountCredential storageCreds;
 
+    @DataBoundSetter
+    public void setIncludeFilesPattern(String includeFilesPattern) {
+        this.includeFilesPattern = includeFilesPattern;
+    }
+
+    @DataBoundSetter
+    public void setExcludeFilesPattern(String excludeFilesPattern) {
+        this.excludeFilesPattern = excludeFilesPattern;
+    }
+
+    @DataBoundSetter
+    public void setDownloadDirLoc(String downloadDirLoc) {
+        this.downloadDirLoc = downloadDirLoc;
+    }
+
+    @DataBoundSetter
+    public void setFlattenDirectories(boolean flattenDirectories) {
+        this.flattenDirectories = flattenDirectories;
+    }
+
+    @DataBoundSetter
+    public void setIncludeArchiveZips(boolean includeArchiveZips) {
+        this.includeArchiveZips = includeArchiveZips;
+    }
+
     public static class DownloadType {
 
         public final String type;
```

Comment:
this is the recommended way by the [pipeline dev guide](https://github.com/jenkinsci/pipeline-plugin/blob/master/DEVGUIDE.md#constructor-vs-setters) to reduce the mandatory params and give appropriate default value.